src/server/scripts/EasternKingdoms/zone_tirisfal_glades.cpp

```diff
@@ -54,82 +60,59 @@ class npc_calvin_montague : public CreatureScript
     {
-        npc_calvin_montagueAI(Creature* creature) : ScriptedAI(creature)
-        {
-            Initialize();
-        }
-
-        void Initialize()
-        {
-            m_uiPhase = 0;
-            m_uiPhaseTimer = 5000;
-            m_uiPlayerGUID.Clear();
-        }
-
-        uint32 m_uiPhase;
-        uint32 m_uiPhaseTimer;
-        ObjectGuid m_uiPlayerGUID;
+        npc_calvin_montagueAI(Creature* creature) : ScriptedAI(creature) { }
 
         void Reset() override
         {
-            Initialize();
-
             me->RestoreFaction();
-
             if (!me->IsImmuneToPC())
                 me->SetImmuneToPC(true);
         }
 
         void EnterCombat(Unit* /*who*/) override { }
 
-        void DamageTaken(Unit* pDoneBy, uint32 &uiDamage) override
+        void DamageTaken(Unit* /*attacker*/, uint32 &damage) override
         {
-            if (uiDamage > me->GetHealth() || me->HealthBelowPctDamaged(15, uiDamage))
+            if (damage > me->GetHealth() || me->HealthBelowPctDamaged(15, damage))
             {
-                uiDamage = 0;
-
-                me->RestoreFaction();
-                me->SetImmuneToPC(true);
+                damage = 0;
                 me->CombatStop(true);
-
-                m_uiPhase = 1;
-
-                if (pDoneBy->GetTypeId() == TYPEID_PLAYER)
-                    m_uiPlayerGUID = pDoneBy->GetGUID();
+                EnterEvadeMode();
+                _events.ScheduleEvent(EVENT_EMOTE_RUDE, Seconds(3));
             }
         }
 
         void UpdateAI(uint32 diff) override
         {
-            if (m_uiPhase)
-            {
-                if (m_uiPhaseTimer <= diff)
-                    m_uiPhaseTimer = 7500;
-                else
-                {
-                    m_uiPhaseTimer -= diff;
-                    return;
-                }
+            _events.Update(diff);
 
-                switch (m_uiPhase)
+            if (uint32 eventId = _events.ExecuteEvent())
+            {
+                switch (eventId)
                 {
-                    case 1:
+                    case EVENT_EMOTE_RUDE:
+                        me->HandleEmoteCommand(EMOTE_ONESHOT_RUDE);
+                        _events.ScheduleEvent(EVENT_TALK, Seconds(2));
+                        break;
+                    case EVENT_TALK:
                         Talk(SAY_COMPLETE);
-                        ++m_uiPhase;
+                        _events.ScheduleEvent(EVENT_DRINK, Seconds(5));
                         break;
-                    case 2:
-                        if (Player* player = ObjectAccessor::GetPlayer(*me, m_uiPlayerGUID))
+                    case EVENT_DRINK:
+                        if (Player* player = ObjectAccessor::GetPlayer(*me, _playerGUID))
                             player->AreaExploredOrEventHappens(QUEST_590);
-
+                        _playerGUID.Clear();
                         DoCast(me, SPELL_DRINK, true);
```

Comment:
DoCastSelf(SPELL_DRINK, true);

Reply:
Removing the triggered parameter altogether, not needed here. Thanks :)